### DISCUSSION TITLE
[2.3.2] Disabling SIDs from minion pillars failing

### DISCUSSION THREAD

**Discussion by chuglo** [Security-Onion-Solutions/securityonion#1781] [Unsupported Versions] [answered]
Hello,

- SO v2.3.2
- Distributed: Manager, Search Node, Sensor

I am attempting to disable a SID from within a minion pillar per https://docs.securityonion.net/en/2.3/managing-alerts.html#disable-the-sid 

I added the following to the end of `/opt/so/saltstack/local/pillar/minions/$hostname_sensor.sls`

```
idstools:
  sids:
    disabled:
      - 2012647
```

but when I apply the salt state, `sudo salt $hostname_sensor state.apply idstools`, I get 

```
$hostname_sensor:
----------
          ID: idstools_state_not_allowed
    Function: test.fail_without_changes
      Result: False
     Comment: Failure!
     Started: 14:57:12.767780
    Duration: 2.575 ms
     Changes:

Summary for $hostname
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   2.575 ms
ERROR: Minions returned with non-zero exit code
```

Any ideas as to why the state is failing to apply? Am I doing something wrong?

Thank you!

**Comment by netsecninja** [comment]
Make sure you are using spaces and not tabs in the .sts file. I learned that the hard way. And just to be sure, you are not using $hostname_sensor in your commands, but replacing it with your proper hostname_role? For example, my host is named securityonion, and the role is standalone, so it would be `sudo salt securityonion_standalone state.apply idstools`

**Comment by chuglo** [comment, marked as answer]
Yes, I am using the actual hostname in my commands. 

It seems the error is due to `idstools` not running on sensors? The only place that I see an `idstools` container is the manager and that is the only place a `state.apply idstools` will run successfully. I removed the entire disable entry that I put in the minion pillar file and tried to run `state.apply` for giggles and got the same error.

>To enable or disable SIDs for Suricata, the Salt idstools pillar can be used in the minion pillar file (/opt/so/saltstack/local/pillar/minions/<minionid>.sls). In a distributed Security Onion environment, you only need to change the configuration in the manager pillar and then all other nodes will get the updated rules automatically.

https://docs.securityonion.net/en/2.3/managing-alerts.html?highlight=threshold#disable-the-sid 

Can the `idstools` pillar actually be used in minion pillar file? and if so, how should it be applied? Right now, it seems we can only disable SIDs fleet-wide.

**Reply by TOoSmOotH** [reply]
This is correct. At the moment SIDS are disabled grid wide. Thresholds might be a better solution here if you need to see an alert on some sensors and not on others. I have added #1802 to allow more granular sid disablement.

**Comment by rwwiv** [comment]
@chuglo  
We're currently in the process of writing a tool to make disabling SIDs easier. If you'd like to test it out early, copy [this link's content](https://raw.githubusercontent.com/Security-Onion-Solutions/securityonion/feature/so-rules/salt/common/tools/sbin/so-rules) into `/usr/sbin/so-rules` on the manager (or alternatively, copy the script from the `feature/so-rules` branch of this repo - found under `securityonion/salt/common/tools/sbin`), then copy the following script somewhere onto the same system and run it as root:
``` sh
#!/bin/bash

. /usr/sbin/so-common

# Surround curly braces in any minion pillar in single quotes so the pillars can be parsed as regular yaml

readarray -t minion_pillars <<< "$(find /opt/so/saltstack/local/pillar/minions -type f -name '*.sls')"
for pillar in "${minion_pillars[@]}"; do
  sed -i -r "s/ (\{\{.*}})$/ '\1'/g" "$pillar" 
done
```
---
This tool still only supports globally disabling SIDs, so if you're looking to disable per-minion then follow @TOoSmOotH's comment and use thresholding.

**Reply by launchdaemon** [reply]
Hi @rwwiv , thanks for the info.  I am having trouble disabling a SID on my standalone security onion 2.3.21.  I will try your suggestion but also wanted to mention that there is a typo in the link you included, I don't think there should be an `s` at the end as `so-rules` does not exist in that repo but `so-rule` does [here](https://raw.githubusercontent.com/Security-Onion-Solutions/securityonion/feature/so-rules/salt/common/tools/sbin/so-rule).
